src/lib.rs

```diff
@@ -963,28 +968,51 @@ impl<'a> Cmd<'a> {
         if !self.data.quiet {
             eprintln!("$ {}", self);
         }
-        self.output_impl(false, false).map(|_| ())
+        self.output_impl(false, false, None).map(|_| ())
+    }
+
+    /// Run the command with a timeout.
+    pub fn run_timeout(&self, timeout: Duration) -> Result<()> {
+        if !self.data.quiet {
+            eprintln!("$ {}", self);
+        }
+        self.output_impl(false, false, Some(timeout)).map(|_| ())
     }
 
     /// Run the command and return its stdout as a string. Any trailing newline or carriage return will be trimmed.
     pub fn read(&self) -> Result<String> {
-        self.read_stream(false)
+        self.read_stream(false, None)
+    }
+
+    /// Run the command with a timeout and return its stdout as a string. Any trailing newline or carriage return will be trimmed.
+    pub fn read_timeout(&self, timeout: Duration) -> Result<String> {
```

Comment:
Rather than adding a separate `xxx_timeout` method for each of operations we have, let's rather add a single `.timeout()` builder method, a-la `ignore_status`